desktop/api/src/main/java/org/datacleaner/actions/PreviewTransformedDataActionListener.java

```diff
@@ -136,12 +168,52 @@ public TableModel call() throws Exception {
             rowCollector.setComponentRequirement(tjb.getComponentRequirement());
         }
 
-        // add a max rows filter
-        final FilterComponentBuilder<MaxRowsFilter, MaxRowsFilter.Category> maxRowFilter = ajb
-                .addFilter(MaxRowsFilter.class);
-        maxRowFilter.setName(getClass().getName() + "-MaxRows");
-        maxRowFilter.getComponentInstance().setMaxRows(_previewRows);
-        ajb.setDefaultRequirement(maxRowFilter, MaxRowsFilter.Category.VALID);
+        // add a max rows filter to the source of the job
+        final AnalysisJobBuilder rootJobBuilder = ajb.getRootJobBuilder();
+        {
+            // remove any existing max rows filters
+            final List<FilterComponentBuilder<?, ?>> filters = new ArrayList<>(
+                    rootJobBuilder.getFilterComponentBuilders());
+            for (FilterComponentBuilder<?, ?> filter : filters) {
+                if (filter.getComponentInstance() instanceof MaxRowsFilter) {
+                    rootJobBuilder.removeFilter(filter);
+                }
+            }
+
+            final SourceColumnFinder sourceColumnFinder = new SourceColumnFinder();
+            sourceColumnFinder.addSources(rootJobBuilder);
+            final List<Table> sourceTables = rootJobBuilder.getSourceTables();
+            final int maxRows = Double.valueOf(Math.ceil(((double) _previewRows) / sourceTables.size())).intValue();
```

Comment:
It's not really a problem currently, but I don't believe that there is anything inherently in the multi stream component base class that makes 1 row per source table row, so we may end up with less rows that actually wanted for a new component, unless the developer is aware of this limitation.

Maybe we should at least point it out in the JavaDoc for the ```MultiStreamComponent``` class.